### PR TITLE
Improve message bus topic matching performance

### PR DESCRIPTION
**Pull request by ryantam626**
# Pull Request

Improve `nautilus_trader.common.component.is_matching` exact match performance as suggested in #2150.

Also adds some more test cases.

## Type of change

Bug fix (non-breaking change which fixes an issue)

## How has this change been tested?

Existing tests are almost sufficient.

For completeness, I have also added some more tests for
1. `?` pattern which we do support
2. `[seq]` and `[!seq]` pattern which do NOT support

### CODE REVIEW

**Comment by ryantam626** [comment]
Actually I am not certain if `topic_contains_wildcard = contains_wildcard(topic)` and its check is required.

I was under the impression that it's required, because of this test https://github.com/nautechsystems/nautilus_trader/blob/c53f87ac40f0907b4313d826c5e0637a30665d01/tests/unit_tests/msgbus/test_bus.py#L463 in particular.

However upon trying to add more test which flips all the existing tests' `topic` and `pattern` around, they all failed, which led me to think that `topic_contains_wildcard = contains_wildcard(topic)` is actually not required - if that's the case then I think this can almost be reduced to a `fnmatch.fnmatch` (well depending on if we want to support other shell-style patterns?)

**Comment by cjdsellers** [comment]
Hey @ryantam626 

There is a chance the implementation is not correct. We only have to preserve the current functionality, and this will be revisited and optimized in Rust later (current impl [here](https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_core/common/src/msgbus/mod.rs)).

**Comment by ryantam626** [comment]
Whoops, forgot to follow the contributor guide, will fix the CI failures.

**Comment by ryantam626** [comment]
Should be fixed now, sorry for the repeated force push 😓 

FWIW, I think it probably is correct - I just got the wrong end of the stick about the purpose of the `["*", "*", True]` test!

I will give this patch a spin and follow up with another PR if I find a better solution in the future.

File: nautilus_trader/common/component.pyx [hunk 1]
```diff
@@ -2596,7 +2596,17 @@ cdef class MessageBus:
         return subs_array
 
 
+cdef inline bint contains_wildcard(str topic_or_pattern):
+    return '?' in topic_or_pattern or '*' in topic_or_pattern
+
+
 cdef inline bint is_matching(str topic, str pattern):
+    topic_contains_wildcard = contains_wildcard(topic)
```
Comment (cjdsellers) [inline nautilus_trader/common/component.pyx:2604]: Small Cython thing: if a local variable isn't typed with a `cdef` then the Python interpreter is needed to figure out the type. Will merge like this though and optimize on `develop`.

**Pull request merged**